rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/common/event/RMEvent.java

```diff
@@ -136,4 +145,27 @@ public String toString() {
     public Object clone() throws CloneNotSupportedException {
         return super.clone();
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+
+        RMEvent rmEvent = (RMEvent) o;
+
+        return counter == rmEvent.counter;
+    }
+
+    @Override
+    public int hashCode() {
+        return (int) (counter ^ (counter >>> 32));
```

Comment:
We could put in a comment the logic behind this hashcode computation.

Reply:
Actually it is IDEA who generated them